applications/asset_tracker_v2/doc/data_module.rst

```diff
@@ -0,0 +1,109 @@
+.. _asset_tracker_v2_data_module:
+
+Data module
+###########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The data module gathers data that have been sampled by other module in the system and stores it into ringbuffers.
+It keeps track of data that have been requested by the :ref:`asset_tracker_v2_app_module` and decides when data is sent cloud.
+
+.. note::
+   The data module will undergo substatial refactoring in the nearest feature. Due to this, a some of its features are not currently documented.
+
+Features
+********
+
+This section documents the various features implemented by the module.
+
+Requested data types
+====================
+
+The data module maintans a list of requested data types that is updated every time a sample request of type :c:enum:`APP_EVT_DATA_GET` is sent from the :ref:`asset_tracker_v2_app_module`.
+When all the data types present in the aforementioned list have been received by module the event :c:enum:`DATA_EVT_DATA_READY` is sent out.
+This event is a signal that all data has been gathered for a particlar sample request and that it can be encoded and sent to cloud.
+The :c:enum:`APP_EVT_DATA_GET` event also carries a sample timeout. This timeout is used by the data module to trigger a :c:enum:`DATA_EVT_DATA_READY` event, even if not all types in the requested list have been receieved.
+This acts as a failsafe in case some module that data is expected from malfunctions or does not manage to sample the requested data in time.
+
+.. note::
+   The module can decide to not send data after a sample request. If this happens, data is still persisted in the ringbuffers and will be sent to cloud in batch messages after the next sample request given that the application is connected to cloud.
+   The ringbuffers in the module are implemented so that the oldest entry will always be overwritten in case the buffer is filled.
+
+Device configuration
+====================
+
+Upon boot, the data module distributes the current values of the applications :ref:`Real-time configurations <real_time_configs>` to the rest of the application with the :c:enum`DATA_EVT_CONFIG_INIT` event.
+These value are persisted in flash between reboots using the :ref:`settings_api` and updated each time the application receives a new configuration update.
+If a new configuration update is received from cloud, the data module distributes the new configurations values via the :c:enum:`DATA_EVT_CONFIG_READY` event.
+The default values of the :ref:`Real-time configurations <real_time_configs>` can be altered compile time using the compile time options listed in :ref:`Default device configuration options <default_config_values>`.
+
+Configuration options
+*********************
+
+.. _default_config_values:
+
+.. _CONFIG_DATA_DEVICE_MODE:
+
+CONFIG_DATA_DEVICE_MODE - Configuration for the device mode
+   This configuration sets the device mode.
```

Comment:
```suggestion
CONFIG_DATA_DEVICE_MODE - This configuration sets the device mode.
```

Reply:
These descriptions and below fix the repetition repetition. The sentence below does not contain additional information and can be removed for brevity.